mlx/directives/item_pie_chart_directive.py

```diff
@@ -198,25 +199,11 @@ def run(self):
         if self.arguments:
             item_chart_node['title'] = self.arguments[0]
 
-        # Process ``id_set`` option
-        if 'id_set' in self.options and len(self.options['id_set']) >= 2:
-            item_chart_node['id_set'] = self.options['id_set'].split()
-        else:
-            item_chart_node['id_set'] = []
-            report_warning(env, 'Traceability: Expected at least two arguments in id_set.', env.docname, self.lineno)
+        self._process_id_set(item_chart_node, env)
 
-        # Process ``label_set`` option
-        default_labels = ['uncovered', 'covered', 'executed']
-        if 'label_set' in self.options:
-            item_chart_node['label_set'] = [x.strip(' ') for x in self.options['label_set'].split(',')]
-            if len(item_chart_node['label_set']) != len(item_chart_node['id_set']):
-                item_chart_node['label_set'].extend(
-                    default_labels[len(item_chart_node['label_set']):len(item_chart_node['id_set'])])
-        else:
-            id_amount = len(item_chart_node['id_set'])
-            item_chart_node['label_set'] = default_labels[:id_amount]  # default labels
+        self._process_label_set(item_chart_node)
 
-        # Add found attribute to item. Attribute data is a comma-separated list of strings
+        self._process_attribute(item_chart_node, env)
         item_chart_node['attribute'] = ''
         item_chart_node['priorities'] = []
         for attr in TraceableItem.defined_attributes.keys():
```

Comment:
I will assume this here needs to be removed (whole for loop)

Reply:
absolutely 583a4ef66f6438d8a5f21b57a3fc233993aeb473